pyne/vta.py

```diff
@@ -0,0 +1,481 @@
+    -----------
+    start : numpy array 
+        An array of size 3, represents the start point of  the line segment
+    end : numpy array
+        An array of size 3, represents the end point of  the line segment
+
+    Return:
+    -------
+    dist : float
+        The distance between the start point and the end point.
+    """
+    dist = math.sqrt((end[0] - start[0]) * (end[0] - start[0]) +
+                     (end[1] - start[1]) * (end[1] - start[1]) +
+                     (end[2] - start[2]) * (end[2] - start[2]))
+    return dist
+
+def _is_point_in_mesh(bounds, point):
+    """
+    Check whether a point is in the mesh.
+
+    Parameters:
+    -----------
+    bounds : list
+        Boundaries. Nested list of the boundaries along x, y, and z directions.
+        Expected format: [[x_min, ..., x_max],
+                          [y_min, ..., y_max],
+                          [z_min, ..., z_max]]
+    point : numpy array
+        An array with size of 3, represents the point to be checked.
+
+    Return:
+    -------
+    True : when point in the mesh.
+    False : when point outside the mesh.
+    """
+    if bounds[0][0] <= point[0] <= bounds[0][-1] and \
+       bounds[1][0] <= point[1] <= bounds[1][-1] and \
+       bounds[2][0] <= point[2] <= bounds[2][-1]:
+        return True
+    else:
+        return False
+
+def _find_voxel_idx_1d(bound, cor, v_1d):
+    """
+    Find the voxel id (in a given dimension).
+
+    Parameters:
+    -----------
+    bound : list
+        Boundary of the mesh in a given direction.
+    cor : float
+        Coordinate of the point in the given direction.
+    v_1d : float
+        Vector attribute in the given direction.
+
+    Return:
+    -------
+    idx : int
+        Mesh index of the given direction. This value is -1 when the point not
+        in the mesh.
+    ----------
+    """
+    idx = -1
+    if v_1d > 0:
+        for i in range(len(bound)-1):
+            if bound[i] <= cor < bound[i+1]:
+                idx = i
+                break
+    else:
+        for i in range(len(bound)-1, -1, -1):
+            if bound[i] < cor <= bound[i+1]:
+                idx = i
+                break
+    # idx could be -1, that means point outside the mesh
+    return idx
+
+def _cal_dir_v(start, end):
+    """
+    Calculate the direction from start to end (start -> end).
+    
+    Parameters:
+    -----------
+    start : numpy array
+        An array of size 3. The start point.
+    end :
+        An array of size 3. The end point.
+
+    Return:
+    -------
+    v : numpy array
+        Direction from start to end, an unit vector.
+    """
+    v = end - start
+    dist = math.sqrt(v[0] * v[0] + v[1] * v[1] + v[2] * v[2])
+    v = np.divide(v, dist)
+    return v
+
+def _find_next_grid_1d(cor, v_1d, bound):
+    """
+    Find the next grid coordinate of a direction.
+
+    Parameters:
+    -----------
+    cor : float
+        Coordinate of current point in the direction.
+    v_1d : float
```

Comment:
Is there a particular reason this is a float rather than a Bool or Int?

Reply:
The `v_1d` here is a number in `[v_x, v_y, v_z]`, it is a float. It can be converted to a `bool` or `int` before the next `if` function. However, the conversion will not simplify the code. For the better understanding of the code, I keep this parameter to be a float.

Reply:
I think that's ok, yeah. The odd part to me was a float being used to make a binary choice in the code, but I see what you mean. It also comes in handy at the end of the function for setting the sign of INF. 

I'm going to propose something else to simplify the code here though. The for loops can be consolidated like so: 

```
if v_1d < 0: 
    bound = reversed(bound)
    cor = -cor

for bnd in bound:
    if cor < bnd_pos:
        n_grid = bnd_pos
        flag = True
        break
```

The for loop code can always be the same if the iterable is setup correctly.